internal/mempool/mempool.go

```diff
@@ -572,15 +572,23 @@ func (txmp *TxMempool) addNewTransaction(wtx *WrappedTx, res *abci.ResponseCheck
 	sender := res.Sender
 	priority := res.Priority
 
+	// if sender is set, then priority must be higher on the new tx to replace
+	// otherwise this rejects that transaction.  At this time, only EVM transactions
+	// set the sender
```

Comment:
we'd also need to add an sender index in `pendingTxs` and check for existing tx in `pendingTxs`